spec/deploy_bouncer_spec.rb

```diff
@@ -71,5 +71,103 @@
         end
       end
     end
+
+    it "fails when APP_DOMAIN is not set" do
+      ClimateControl.modify FASTLY_SERVICE_ID: "123321abc", FASTLY_API_KEY: "fastly@example.com" do
+        expect { DeployBouncer.new.deploy! }
+        .to raise_error(SystemExit, /APP_DOMAIN environment variable is not set/)
+      end
+    end
+
+    it "fails when FASTLY_SERVICE_ID is not set" do
+      ClimateControl.modify APP_DOMAIN: "gov.uk", FASTLY_API_KEY: "fastly@example.com" do
+        expect { DeployBouncer.new.deploy! }
+        .to raise_error(SystemExit, /FASTLY_SERVICE_ID environment variable is not set/)
+      end
+    end
+
+    it "fails when FASTLY_API_KEY is not set" do
+      ClimateControl.modify APP_DOMAIN: "gov.uk", FASTLY_SERVICE_ID: "123321abc" do
+        expect { DeployBouncer.new.deploy! }
+        .to raise_error(RuntimeError, /FASTLY_API_KEY is not set in the environment/)
+      end
+    end
+
+    it "fails when no hosts found in Transition hosts API" do
+      @requests = []
+
+      # Fastly#get_service. Return a service with two VCL "versions" (https://docs.fastly.com/api/config#version)
+      @requests << stub_request(:get, "https://api.fastly.com/service/123321abc")
+        .to_return(body: File.read("spec/fixtures/fastly-get-service-response.json"))
+
+      # We clone the latest active VCL version, which returns the latest version
+      @requests << stub_request(:put, "https://api.fastly.com/service/123321abc/version/2/clone")
+        .to_return(body: File.read("spec/fixtures/fastly-put-clone.json"))
+
+      # Given Transition has no hosts
+      @requests << stub_request(:get, "https://transition.publishing.service.gov.uk/hosts.json")
+        .to_return(body: JSON.dump(results: []))
+
+      # And Fastly has one host
+      @requests << stub_request(:get, "https://api.fastly.com/service/123321abc/version/3/domain")
+        .to_return(body: JSON.dump([{ name: "existing.example.com" }]))
+
+      ClimateControl.modify APP_DOMAIN: "gov.uk", FASTLY_SERVICE_ID: "123321abc", FASTLY_API_KEY: "fastly@example.com" do
+        expect { DeployBouncer.new.deploy! }
+        .to raise_error(RuntimeError, /No hosts found in Transition hosts API/)
+
+        @requests.each do |request|
+          expect(request).to have_been_requested.at_least_once
+        end
```

Comment:
I think we can use [RSpec's `all` matcher](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/all-matcher) here:

```suggestion
        expect(@requests).to all ( have_been_requested.at_least_once )
```